### PR TITLE
4.0 compatible layouts

### PR DESCRIPTION
**Pull request by chmst**
Pull Request for Issue # .

Backend
- [x]  access.xml
- [x]  config.xml  - adapt format to J4
- [ ]  config.xml  - review images and al-text (a11y), review usage of icon 
- [x]  adminsitrator/forms/weblink.xml  - adapt format to J4
- [x]  adminsitrator/forms/weblinks_filter.xml  - adapt format to J4
- [ ]  adminsitrator/forms/weblink.xml  - review images and al-text (a11y)
- [ ]  adminstrator/languages/com_weblinks.ini (remove unused strings)
- [ ]  language files - clean up
- [x]  weblinks view 
- [x]  weblinks view modal
- [x]  weblinks batch 
- [ ]  tbd usage of language in batch modal
- [x]  weblink edit form layout
- [ ]  weblink edit form - improve image fieldsets
- [ ]  remove layouts associations, metadata params
 
Frontend

to be continued

### CODE REVIEW

**Comment by chmst** [comment]
We probably can remove some files from layouts - but maybe we should deprecate them?

File: src/administrator/components/com_weblinks/tmpl/weblinks/modal.php [hunk 1]
```diff
@@ -14,31 +15,29 @@
...
-HTMLHelper::_('script', 'com_weblinks/admin-weblinks-modal.js', array('version' => 'auto', 'relative' => true));
-HTMLHelper::_('bootstrap.tooltip', '.hasTooltip', array('placement' => 'bottom'));
-HTMLHelper::_('formbehavior.chosen', 'select');
+HTMLHelper::_('behavior.multiselect');
 
-// Special case for the search field tooltip.
-$searchFilterDesc = $this->filterForm->getFieldAttribute('search', 'description', null, 'filter');
-HTMLHelper::_('bootstrap.tooltip', '#filter_search', array('title' => Text::_($searchFilterDesc), 'placement' => 'bottom'));
+/** @var Joomla\CMS\WebAsset\WebAssetManager $wa */
+$wa = $this->document->getWebAssetManager();
```
Comment (joomdonation) [inline src/administrator/components/com_weblinks/tmpl/weblinks/modal.php:31]: Change these lines of code to below code should fix the error you are having:

```php
$wa = $this->document->getWebAssetManager()
	->registerAndUseScript('com_weblinks.admin-weblinks-modal', 'media/com_weblinks/js/admin-weblinks-modal.js', [], ['defer' => true], ['core']);
```

File: src/administrator/components/com_weblinks/tmpl/weblink/edit.php [hunk 2]
```diff
@@ -55,33 +59,47 @@
...
-					<?php foreach ($this->form->getGroup('images') as $field) : ?>
-						<?php echo $field->renderField(); ?>
-					<?php endforeach; ?>
-				</div>
+		<div class="row">
+			<div class="col-12">
+				<fieldset id="fieldset-image; ?>" class="options-form">
+					<legend><?php echo Text::_('JGLOBAL_FIELDSET_IMAGE_OPTIONS'); ?></legend>
+					<div>
+						<?php echo $this->form->renderField('imaJGLOBAL_FIELDSET_IMAGE_OPTIONSges'); ?>
```
Comment (brianteeman) [inline src/administrator/components/com_weblinks/tmpl/weblink/edit.php:67]:
```suggestion
						<?php echo $this->form->renderField('images'); ?>
```

File: src/administrator/components/com_weblinks/tmpl/weblinks/default_batch_body.php [hunk 3]
```diff
@@ -8,33 +8,36 @@
...
+	<div class="row">
+		<?php if (Multilanguage::isEnabled()) : ?>
+			<div class="form-group col-md-6">
+				<div class="controls">
+					<?php echo LayoutHelper::render('joomla.html.batch.language', []); ?>
+				</div>
 			</div>
-		</div>
-		<div class="control-group span6">
+		<?php endif;?>
```
Comment (brianteeman) [inline src/administrator/components/com_weblinks/tmpl/weblinks/default_batch_body.php:25]:
```suggestion
		<?php endif; ?>
```

Comment (chmst) [reply src/administrator/components/com_weblinks/tmpl/weblinks/default_batch_body.php]: thx

File: src/administrator/components/com_weblinks/tmpl/weblinks/modal.php (hunk 1 above)
Comment (chmst) [reply src/administrator/components/com_weblinks/tmpl/weblinks/modal.php]: thx

File: src/administrator/components/com_weblinks/tmpl/weblink/edit.php (hunk 2 above)
Comment (chmst) [reply src/administrator/components/com_weblinks/tmpl/weblink/edit.php:67]: thx

**Pull request merged**